src/bin/dhcp6/dhcp6_srv.cc

```diff
@@ -3818,12 +3849,88 @@ Dhcpv6Srv::requestedInORO(const Pkt6Ptr& query, const uint16_t code) const {
     return (false);
 }
 
+void Dhcpv6Srv::updateHostKey(AllocEngine::ClientContext6& ctx) {
+    //Default arguments generates a random key
+    AuthKey default_key;
+
+    //check if reservation is present for the client
+    if (ctx.hosts_.empty()) {
+        //create and assign new reservation
+        HostPtr hptr(new Host(ctx.duid_->toText(), "duid",
+                     SubnetID(1), ctx.subnet_->getID(),
+                     IOAddress::IPV4_ZERO_ADDRESS()));
+
+        hptr->setKey(default_key);
+
+        //we are creating new host reservation
+        //ensure to add only if a backend is configured
+        //we dont want to enforce enabling backend 
+        if(!HostMgr::instance().getHostDataSourceList().empty()) { 
```

Comment:
There is a design issue here: we do not want of course to enforce to have a host backend but on the other hand we want the reconfigure to work even there is no host backend... More in the main (i.e. not file/line attached) conversation thread.